contracts/interfaces/base/IStorage.sol

```diff
@@ -31,15 +31,23 @@ import { CallType } from "../../lib/ModeLib.sol";
 interface IStorage {
     /// @notice Struct storing validators and executors using Sentinel lists, and fallback handlers via mapping.
     struct AccountStorage {
-        SentinelListLib.SentinelList validators; ///< List of validators, initialized upon contract deployment.
-        SentinelListLib.SentinelList executors; ///< List of executors, similarly initialized.
-        mapping(bytes4 => FallbackHandler) fallbacks; ///< Mapping of selectors to their respective fallback handlers.
-        IHook hook; ///< Current hook module associated with this account.
+        SentinelListLib.SentinelList validators;
+        ///< List of validators, initialized upon contract deployment.
+        SentinelListLib.SentinelList executors;
+        ///< List of executors, similarly initialized.
+        mapping(bytes4 => FallbackHandler) fallbacks;
+        ///< Mapping of selectors to their respective fallback handlers.
+        IHook hook;
+        ///< Current hook module associated with this account.
+        mapping(address hook => uint256) emergencyUninstallTimelock;
```

Comment:
do we provide emergencyUninstallTimelock while installing hook now?
mapping makes me think multiple hooks associated with uninstall timelocks

Reply:
so how it works is that you can initiate a timelock for a hook and when that timelock is expired you can uninstall the hook. It uses a mapping since you could initiate the timelock, then uninstall the hook normally, reinstall it and then still have the timelock counting down even though its a different hook now